src/components/ZipSearchField.jsx

```diff
@@ -0,0 +1,50 @@
+export default function ZipSearchField(props) {
+
+    const zipcodeApi = async (zipcode) => {
+        // put try catch block here
+        // API_BASE_URL specifies the API Base URL
+        const API_BASE_URL = "https://ctp-zip-code-api.onrender.com";
+        //try block to test if api call gives valid response or error
+        try{
+            // using a concatonated string to create the APU URL and then use fetch to call the API
+            const response = await fetch(API_BASE_URL + "/zip/" + zipcode);
+            // var body will store the reponses
+            const body = await response.json();   
+            // we passed the setCityList function so we can call the prop function to set the city list
+            props.setCityList(body);
+            // if an error is caught change the cityList to an empty array and show the "results not found text"
+        }catch(error){
+            document.getElementById("no-results").style.display = "block";
```

Comment:
In React, we normally avoid manipulating DOM nodes. We allow React to do it instead.